### PR TITLE
MINOR: Add a duplicate() method to Message classes

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java [hunk 1]
```diff
@@ -2112,6 +2131,83 @@ private void generateFieldHashCode(FieldSpec field) {
...
+            } else if (field.type().isString()) {
+                // Strings are immutable, so we don't need to duplicate them.
+                cond.ifShouldNotBeNull(() ->
+                    buffer.printf("%s;%n", target.assignmentStatement(
+                        target.sourceVariable())));
+            } else if (field.type().isArray()) {
+                cond.ifShouldNotBeNull(() -> {
+                    String newArrayName =
+                        String.format("new%s", field.capitalizedCamelCaseName());
+                    buffer.printf("%s %s = new %s();%n",
```
Comment (chia7712) [inline generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java]: What about adding the expected number of elements?

Comment (cmccabe) [reply generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java]: Good idea.  Added.

File: generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java [hunk 2]
```diff
@@ -273,6 +276,22 @@ private String commaSeparatedHashSetFieldAndTypes(StructSpec struct) {
             collect(Collectors.joining(", "));
     }
 
+    private void generateDuplicateMethod(String className, StructSpec struct) {
```
Comment (hachikuji) [inline generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java]: nit: maybe `generateCollectionDuplicateMethod`?

**Review by hachikuji — APPROVED** [review]
LGTM. Left a minor naming suggestion. It would be nice if we could do something like `testDuplication` for each of the generated API types as well.

**Pull request merged**